pennylane_lightning/core/src/simulators/lightning_kokkos/gates/GateFunctorsParam.hpp

```diff
@@ -1687,4 +1483,421 @@ template <class PrecisionT, bool inverse = false> struct rotFunctor {
+    void operator()(const std::size_t k) const {
+        const std::size_t kdim = k * dim;
+        INDEX(i000, 0B000);
+        ComplexT v000 = arr(i000);
+        INDEX(i001, 0B001);
+        ComplexT v001 = arr(i001);
+        INDEX(i010, 0B010);
+        ComplexT v010 = arr(i010);
+        INDEX(i011, 0B011);
+        ComplexT v011 = arr(i011);
+        INDEX(i100, 0B100);
+        ComplexT v100 = arr(i100);
+        INDEX(i101, 0B101);
+        ComplexT v101 = arr(i101);
+        INDEX(i110, 0B110);
+        ComplexT v110 = arr(i110);
+        INDEX(i111, 0B111);
+        ComplexT v111 = arr(i111);
+
+        arr(i000) = GATESUM3(0B000);
+        arr(i001) = GATESUM3(0B001);
+        arr(i010) = GATESUM3(0B010);
+        arr(i011) = GATESUM3(0B011);
+        arr(i100) = GATESUM3(0B100);
+        arr(i101) = GATESUM3(0B101);
+        arr(i110) = GATESUM3(0B110);
+        arr(i111) = GATESUM3(0B111);
+    }
+};
+
+#define GATEENTRY4(xx, yy) xx << 4 | yy
+#define GATETERM4(xx, yy, vyy) matrix(GATEENTRY4(xx, yy)) * vyy
+#define GATESUM4(xx)                                                           \
+    GATETERM4(xx, 0B0000, v0000) + GATETERM4(xx, 0B0001, v0001) +              \
+        GATETERM4(xx, 0B0010, v0010) + GATETERM4(xx, 0B0011, v0011) +          \
+        GATETERM4(xx, 0B0100, v0100) + GATETERM4(xx, 0B0101, v0101) +          \
+        GATETERM4(xx, 0B0110, v0110) + GATETERM4(xx, 0B0111, v0111) +          \
+        GATETERM4(xx, 0B1000, v1000) + GATETERM4(xx, 0B1001, v1001) +          \
+        GATETERM4(xx, 0B1010, v1010) + GATETERM4(xx, 0B1011, v1011) +          \
+        GATETERM4(xx, 0B1100, v1100) + GATETERM4(xx, 0B1101, v1101) +          \
+        GATETERM4(xx, 0B1110, v1110) + GATETERM4(xx, 0B1111, v1111)
+
+template <class PrecisionT> struct apply4QubitOpFunctor {
+    using ComplexT = Kokkos::complex<PrecisionT>;
+    using KokkosComplexVector = Kokkos::View<ComplexT *>;
+    using KokkosIntVector = Kokkos::View<std::size_t *>;
+
+    KokkosComplexVector arr;
+    KokkosComplexVector matrix;
+    KokkosIntVector wires;
+    const std::size_t n_wires = 4;
+    const std::size_t dim = static_cast<std::size_t>(1U) << n_wires;
+    std::size_t num_qubits;
+
+    apply4QubitOpFunctor(KokkosComplexVector &arr_, std::size_t num_qubits_,
+                         const KokkosComplexVector &matrix_,
+                         const std::vector<std::size_t> &wires_) {
+        Kokkos::View<const size_t *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>
+            wires_host(wires_.data(), wires_.size());
+        Kokkos::resize(wires, wires_host.size());
+        Kokkos::deep_copy(wires, wires_host);
+        arr = arr_;
+        matrix = matrix_;
+        num_qubits = num_qubits_;
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const std::size_t k) const {
+        const std::size_t kdim = k * dim;
+        INDEX(i0000, 0B0000);
+        ComplexT v0000 = arr(i0000);
```

Comment:
Is there a simple way to make it a single-line statement like:`` ComplexT v0001 = INDEX(0B0001);``?
I think this could improve the reading here.

Reply:
INDEX is a macro that declares and computes `std::size_t i0000`. `i0000` is used to create a copy `v000` of `arr(i000)` and then used to reassign `arr(i000)` as `GATESUM3(0B000)`. I could modify the macro to have lines like
```
std::size_t i000 = INDEX(i000, 0B000);
```
Is that better?

Reply:
Yes, I got how the macros work, and I think that if you can change it to make these statements single-line would be great.
Thank you for that.